build.gradle.kts

```diff
@@ -1,37 +1,71 @@
-import org.jetbrains.intellij.tasks.PatchPluginXmlTask
-import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
 
 plugins {
     java
     kotlin("jvm") version "1.9.22"
-    id("org.jetbrains.intellij") version "1.16.1"
+    id("org.jetbrains.intellij.platform") version "2.3.0"
 }
 
 group = "eu.theblob42.idea.whichkey"
 version = "0.10.3"
 
 repositories {
     mavenCentral()
+    intellijPlatform {
+        defaultRepositories()
+    }
 }
 
 dependencies {
     implementation("org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.4.21")
     implementation("org.jetbrains.kotlinx:kotlinx-coroutines-jdk8:1.6.4")
 
     testImplementation("junit", "junit", "4.12")
+
+    intellijPlatform {
+        intellijIdeaCommunity("2023.3.3")
+        pluginVerifier()
+        plugins("IdeaVIM:2.10.0")
+    }
 }
 
-tasks.withType<KotlinCompile> {
-    kotlinOptions.jvmTarget = "17"
+intellijPlatform {
+    pluginConfiguration {
+        ideaVersion {
```

Comment:
this is a copy/paste from `ideavim` and `ideavim-easymotion`. Maybe you can check if the metadata is still correctly set after building the plugin.